optimum/fx/quantization.py

```diff
@@ -0,0 +1,232 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Type, Union
+
+import torch
+from torch.fx.node import Argument, Node, Target
+from torch.nn.intrinsic import _FusedModule
+from torch.quantization.fx.graph_module import GraphModule, ObservedGraphModule
+from torch.quantization.quantize_fx import Scope, ScopeContextManager
+from torch.quantization.quantize_fx import fuse_fx as orig_fuse_fx
+from torch.quantization.quantize_fx import prepare_fx as orig_prepare_fx
+from torch.quantization.quantize_fx import prepare_qat_fx as orig_prepare_qat_fx
+from transformers.utils.fx import HFTracer, check_if_model_is_supported, get_concrete_args, symbolic_trace
+
+from .utils import check_if_available
+
+
+if TYPE_CHECKING:
+    from torch.fx import Graph
+    from transformers import PreTrainedModel
+
+
+class QuantizationTracer(HFTracer):
+    specialized_concrete_args: Optional[Dict[str, Any]] = None
+
+    def __init__(self, skipped_module_names: List[str], skipped_module_classes: List[Callable]):
+        super().__init__()
+        self.skipped_module_names = skipped_module_names
+        self.skipped_module_classes = skipped_module_classes
+        # NB: initialized the module_type of top level module to None
+        # we are assuming people won't configure the model with the type of top level
+        # module here, since people can use "" for global config
+        # We can change this if there is a use case that configures
+        # qconfig using top level module type
+        self.scope = Scope("", None)
+        self.node_name_to_scope: Dict[str, Tuple[str, type]] = {}
+
+    def is_leaf_module(self, m: torch.nn.Module, module_qualified_name: str) -> bool:
```

Comment:
nit: I think it would be good to avoid using single-letter variable names in the codebase if possible. would something like the following work?

```suggestion
    def is_leaf_module(self, module: torch.nn.Module, module_qualified_name: str) -> bool:
```

Reply:
It's basically from PyTorch codebase, and it was to keep the same signature, I can change that.